src/wallet/wallet.cpp

```diff
@@ -1957,6 +1957,20 @@ void CWallet::AvailableCoins(vector<COutput>& vCoins, bool fOnlyConfirmed, const
             if (nDepth == 0 && !pcoin->InMempool())
                 continue;
 
+            LockPoints lp;
+            CTxMemPoolEntry entry(*pcoin, 0, 0, 0, 0, false, 0, false, 0, lp);
```

Comment:
Since we've already checked that pcoin is in the mempool, we don't need to do any of this -- we can just use the cached values.  (I think the only way the calculation you do below can fail is if the limits are violated during a reorg.)